sql/analyzer/pushdown.go

```diff
@@ -949,3 +457,53 @@ func pushdownFixIndices(a *Analyzer, n sql.Node, scope *Scope) (sql.Node, transf
 	}
 	return FixFieldIndexesForExpressions(a, n, scope)
 }
+
+// pushdownFiltersToTable attempts to push filters to tables that can accept them
+// TODO not called anywhere, maybe deprecated
```

Comment:
Remove this and other code it calls that isn't used elsewhere

Mark sql.FilteredTable as deprecated